src/Yoti.Auth/Activity.cs

```diff
@@ -37,122 +42,129 @@ public ActivityDetails HandleSuccessfulResponse(AsymmetricCipherKeyPair keyPair,
                     keyPair);
 
-                var profile = new YotiUserProfile
+                _yotiUserProfile = new YotiUserProfile
                 {
                     Id = parsedResponse.receipt.remember_me_id
                 };
 
-                AddAttributesToProfile(attributes, profile);
+                AddAttributesToProfile(attributes);
 
                 return new ActivityDetails
                 {
                     Outcome = ActivityOutcome.Success,
-                    UserProfile = profile
+                    UserProfile = _yotiUserProfile
                 };
             }
         }
 
-        private static void AddAttributesToProfile(AttributeList attributes, YotiUserProfile profile)
+        private static void AddAttributesToProfile(AttributeList attributes)
         {
             foreach (AttrpubapiV1.Attribute attribute in attributes.Attributes)
             {
-                byte[] data = attribute.Value.ToByteArray();
-                switch (attribute.Name)
-                {
-                    case "selfie":
+                YotiAttribute<object> yotiAttribute = AttributeConverter.ConvertAttribute(attribute);
 
-                        switch (attribute.ContentType)
+                PropertyInfo propertyInfo = GetProfilePropertyByProtobufName(yotiAttribute.GetName());
+                string stringValue = yotiAttribute.GetStringValue();
+                var byteValue = Conversion.UtfToBytes(stringValue);
+                switch (attribute.ContentType)
+                {
+                    case ContentType.String:
+                        if (attribute.Name == YotiConstants.AttributeStructuredPostalAddress
+                             && propertyInfo.IsDefined(typeof(IsJsonAttribute)))
                         {
-                            case ContentType.Jpeg:
-                                profile.Selfie = new Image
-                                {
-                                    Type = ImageType.Jpeg,
-                                    Data = data,
-                                    Base64URI = "data:image/jpeg;base64," + Convert.ToBase64String(data)
-                                };
-                                break;
-
-                            case ContentType.Png:
-                                profile.Selfie = new Image
-                                {
-                                    Type = ImageType.Png,
-                                    Data = data,
-                                    Base64URI = "data:image/png;base64," + Convert.ToBase64String(data)
-                                };
-                                break;
+                            var structuredPostalAddressAttributeValue = new YotiAttributeValue(TypeEnum.Json, byteValue);
+                            var structuredPostalAddressAttribute = new YotiAttribute<Dictionary<string, string>>(YotiConstants.AttributeStructuredPostalAddress, structuredPostalAddressAttributeValue);
```

Comment:
I've seen that C# does not support wildcards in generics, but it is probably safe to turn this in a Dictionary<string, Object>

Reply:
Yes, I created a .ToJson() method which used Dictionary<string, Object>, but forgot to update this, will do that next